packages/reports/addon/components/dimension-selector.js

```diff
@@ -108,59 +94,48 @@ export default class DimensionSelector extends Component {
    */
   @computed('allTimeGrains', 'allDimensions')
   get listItems() {
-    let timeGrains;
-
-    if (featureFlag('enableRequestPreview')) {
-      // only option is to add the default time grain (if none is selected)
-      timeGrains = [
-        {
-          name: 'Date Time',
-          category: 'Date',
-          dateTimeDimension: true
-        }
-      ];
-    } else {
-      timeGrains = this.allTimeGrains;
-    }
-
-    return [...timeGrains, ...this.allDimensions];
+    return [
+      {
+        name: 'Date Time',
+        category: 'Date',
+        dateTimeDimension: true
+      },
+      ...this.allDimensions
+    ];
   }
 
   /*
    * @property {Array} selectedDimensions - dimensions in the request
    */
-  @mapBy('request.dimensions', 'dimension')
-  selectedDimensions;
+  @computed('request.columns.[]')
+  get selectedDimensions() {
+    return this.request.columns.filter(IS_DIMENSION);
+  }
 
   /*
    * @property {Array} selectedFilters - filters in the request
    */
   @computed('request.filters.[]')
   get selectedFilters() {
-    return get(this, 'request.filters').mapBy('dimension');
+    return this.request.filters.filter(IS_DIMENSION);
   }
 
   /*
    * @property {Object} selectedTimeGrain - timeGrain in the request
    */
-  @computed('request.logicalTable.timeGrain', 'allTimeGrains')
+  @computed('request.timeGrain', 'currentTable.timeGrains')
   get selectedTimeGrain() {
-    const {
-      allTimeGrains,
-      request: {
-        logicalTable: { timeGrain }
-      }
-    } = this;
-
-    return allTimeGrains.find(grain => grain.id === timeGrain);
+    const { currentTable, request } = this;
```

Comment:
do we still need this with request preview enabled?

Reply:
https://github.com/yahoo/navi/blob/178c82dc1b9b1cd353f055376b37c6cbcda3615f/packages/reports/addon/templates/components/dimension-selector.hbs#L18
It's used in this block, so once we better integrate timeDimension we could fix it